otherlibs/unix/link.c

```diff
@@ -18,18 +18,33 @@
 #include <caml/signals.h>
 #include "unixsupport.h"
 
-CAMLprim value unix_link(value path1, value path2)
+#include <fcntl.h>
+#include <unistd.h>
+
+CAMLprim value unix_link(value follow, value path1, value path2)
 {
-  CAMLparam2(path1, path2);
+  CAMLparam3(follow, path1, path2);
   char * p1;
   char * p2;
   int ret;
+  int flags =
+    Is_block(follow) && Bool_val(Field(follow, 0)) /* Some true */
+    ? AT_SYMLINK_FOLLOW
+    : 0;
   caml_unix_check_path(path1, "link");
```

Comment:
This `int flags` computation should be moved to the `else` part below, and protected by the `#if linkat_is_available`, since `AT_SYMLINK_FOLLOW` may not be defined.